biggraphite/cli/command_copy.py

```diff
@@ -119,14 +119,36 @@ def _copy_metric(accessor, src_metric, dst_metric, time_start, time_end):
         """Copy a metric."""
         log.info("Copying points from '%s' to '%s'" % (src_metric.name, dst_metric.name))
         for stage in src_metric.retention.stages:
+            aggregated_stage = stage.aggregated()
             rounded_time_start = stage.round_up(time_start)
             rounded_time_end = stage.round_up(time_end)
 
+            query_results = accessor.fetch_points(
+                src_metric, rounded_time_start, rounded_time_end, stage, raw=True)
+
+            # Parse points
+            ts_to_pts = {}
+            for successful, rows_or_exception in query_results:
+                if not successful:
+                    continue
+                for row in rows_or_exception:
+                    if aggregated_stage:
+                        # No need to fetch the shard/replica since we will
+                        # select the value with the biggest count for each timestamp
+                        (time_start_ms, offset, _, value, count) = row
+                    else:
+                        (time_start_ms, offset, value) = row
+                        count = 1
+                    timestamp = (time_start_ms + offset * stage.precision_ms) / 1000.0
+
+                    _, current_count = ts_to_pts.get(timestamp, (None, -1))
+                    if count > current_count:
```

Comment:
That doesn't work, you need to group that by replica id.
This field contains both the replica id and the writer id, and once you've choosen a replica you need to get *all* the points for this replica. (you'll get a multiple values for the same repliacas because of carbon restarts)